app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/stream.rb

```diff
@@ -105,6 +107,21 @@ def fetch_deployment_availabilities(parser)
     end
   end
 
+  def fetch_domain_availabilities(parser)
+    fetch_entities_availabilities(parser, @ems.middleware_domains.reload) do |item, avail|
+      status = parser.process_deployment_availability(avail)
```

Comment:
Probably, you could make an alias of this method. It's weird to see `process_deployment_availability` for the domain case.

Although, I would create a new method. This one will return Enabled/Disabled/Unknown. But for domains I think is much better to have Running/Stopped/Unknown, given that domains cannot be "enabled/disabled", but they can be started/stopped.

Reply:
Super! done :)